event-digest/main.go

```diff
@@ -111,45 +142,100 @@ func lineCounter(r io.Reader) (int, error) {
 	return count, nil
 }
 
+func usernameExtractor(r io.Reader, users UsernameSet) error {
+	decoder := json.NewDecoder(r)
+	for {
+		if !decoder.More() {
+			break
+		}
```

Comment:
I'm stupid. Should be `for decoder.More()` -- I forgot you can actually just have a single "clause" boolean `for` loop. https://tour.golang.org/flowcontrol/3